pagefaultgames/pokerogue
# [Optimization] Use inline assets for images/atlases

## What are the changes?

UI Assets that are always loaded inside `LoadingScene` are now loaded as so called `inline-assets`.

[Static Asset Handling | Vite](https://vitejs.dev/guide/assets) 

## Why am I doing these changes?

Using inline assets increases the build-size but improves performance. Since these files are always loaded anyway they can be part of the build-file

## What did change?

I've added a new `loadImageAsset()` and `loadAtlasAsset()` method to load these inline-assets. Then I created some `ts` files that import images & jsons and export them as js object which is then looped through to load the files into phaser.

## Screenshots/Videos

### Network load comparison

#### **With** inline-assets
<img width="416" alt="image" src="https://github.com/user-attachments/assets/988fffa2-d78f-47b8-aa12-72687e4ec348">

#### **Without** inline assets
<img width="434" alt="image" src="https://github.com/user-attachments/assets/45414cc8-1d28-4b0f-b47c-6e970b33ee60">


## How to test the changes?

1. Build the project => `npm run build`
2. Start in `preview` mode => `npm run preview`
3. Open in browser with dev-tools in `Network` tab.
    - Check the amount of requests sent vs. on live/main branch
    - To compare loading times you have to run both locally to have equal chances 
5. Check that everything is still running properly.

## Checklist
